pkg/webhook/volumes.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+Copyright 2024 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	gcsFuseCsiDriverName = "gcsfuse.csi.storage.gke.io"
+)
+
+func (si *SidecarInjector) isGcsFuseCSIVolume(volume corev1.Volume, namespace string) (bool, map[string]string, error) {
+	// Check if it is ephemeral volume.
+	if volume.CSI != nil {
+		if volume.CSI.Driver == gcsFuseCsiDriverName {
+			return true, volume.CSI.VolumeAttributes, nil
```

Comment:
what about csi ephemeral volume which sets the "_" volume handle. that should be skipped as well for metadata prefetch

Reply:
Dynamic mount is now checked inside `isGcsFuseVolume` for both ephemeral and PV.